bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -1,10 +1,35 @@
 # frozen_string_literal: true
 
+require "dependabot/shared_helpers"
+
 module Dependabot
   module Bundler
     module NativeHelpers
-      def self.helper_path
-        "bundle exec ruby #{File.join(native_helpers_root, 'run.rb')}"
+      def self.run_bundler_subprocess(function:, args:, bundler_version:)
+        SharedHelpers.run_helper_subprocess(
+          command: helper_path(bundler_version: bundler_version),
+          function: function,
+          args: args,
+          # Kernel#spawn: clear environment variables except specified by env
+          unsetenv_others: true,
+          env: {
+            # Bundler will pick the matching installed major version
+            "BUNDLER_VERSION" => bundler_version,
+            # Required to find the ruby bin
+            "PATH" => ENV["PATH"],
+            # # Requried to create tmp directories in a writeable folder
+            "HOME" => ENV["HOME"],
+            # Required to git clone to a writeable folder
+            "GEM_HOME" => ENV["GEM_HOME"],
+            # Required by git fetch
+            "SSH_AUTH_SOCK" => ENV["SSH_AUTH_SOCK"]
+          }
+        )
+      end
+
+      def self.helper_path(bundler_version:)
+        native_helper_version = "v#{bundler_version}"
+        "ruby #{File.join(native_helpers_root, native_helper_version, 'run.rb')}"
```

Comment:
👍  intentionally removing as `bundle exec` needs to be run in a folder with a `Gemfile` source (or point to one with a env var) that has resolved dependencies, we currently don't rely on any dependencies in the native helpers so this just got in the way of running the native helpers with a custom bundler version. Bundler refuses to run `BUNDLER_VERSION=1 cmd` if the `Gemfile.lock` that we're pointing to has been resolved with `BUNDLED WITH 2.x` and vice versa.

Reply:
Previously, when we executed `bundle exec cmd` it inherited all existing `ENV` values so ended up pointing to  `./bundler/Gemfile` as we ran the specs with `bundle exec rspec spec` and inherited the same `BUNDLER_VERSION` `ENV` var which I couldn't re-set without setting `unsetenv_others: true`.

Similarly, when run from the updater it would inherit the `BUNDLED WITH` from the updaters `Gemfile.lock`: https://github.com/dependabot/dependabot-updater/blob/main/bin/run#L21